irony-cdb-json.el

```diff
@@ -181,10 +181,26 @@ even helm by enabling `helm-mode' before calling the function."
                    irony-cdb-search-directory-list)
       (expand-file-name it))))
 
+(defvar irony-cdb-json--cache-file nil
```

Comment:
I'll give it a whirl. I'm not sure it will make the code easier to read though. The check will be easier to read, but the let bindings will become more complex.

The real reason I did it this way is the modification time. The way it currently is means it just compared whatever the system can give us as modification time (meaning whatever precision it wants to operate at).

I'll combine the two by appending the modification time as a ISO-8601 string to the filename.

Reply:
I was thinking something like:

```elisp
(defun irony-cdb-json--make-cache-key (file)
  (irony--aif (file-attributes file)
    (cons file (nth 5 it))))
```

Reply:
Nifty. Elisp is not in any way my native language. Thanks. I've updated the code accordingly.